### PR TITLE
[DesktopGL] General Fixes

### PR DESCRIPTION
**Pull request by cra0zy**
Stuff done:
 - Window will no longer get centered after it was moved, this mimics XNA behavior (DesktopGL version of #5467)
 - GL attributes are set before window is created (A proper CreateWindow function when compared to #5445, also I made sure to make it so it can be reused for RecreateWindow with minimal changes later on)

### CODE REVIEW

**Comment by KonajuGames** [comment]
This replaces #5445?  I'll close it when confirmed.

**Comment by cra0zy** [comment]
> This replaces #5445? I'll close it when confirmed.

Yours offers NUnit tests which are useful.

**Comment by KonajuGames** [comment]
If those unit tests are merged into this one, I'm happy to go with this one.

**Comment by Jjagg** [comment]
Just a reminder of what I said in #5445 

> #5429 makes the DesktopGL platform use the new GraphicsDeviceManager, so the changes in GraphicsDeviceManager.Legacy in this PR would need to be moved to PlatformInitialize (called here) in a new file GraphicsDeviceManager.OpenGL when #5429 is merged. Probably easier to get #5429 merged first and then make the necessary changes here.

**Comment by KonajuGames** [comment]
With #5429 merged, this will need to be refactored as mentioned by @Jjagg above.

**Comment by cra0zy** [comment]
@KonajuGames done.

File: Build/Projects/MonoGame.Framework.definition [hunk 1]
```diff
@@ -307,6 +307,9 @@
     <Compile Include="IDrawable.cs" />
     <Compile Include="IGameComponent.cs" />
     <Compile Include="IGraphicsDeviceManager.cs" />
+    <Compile Include="IGraphicsDeviceManager.SDL.cs">
```
Comment (KonajuGames) [inline Build/Projects/MonoGame.Framework.definition]: I think that should have the "I" prefix.

**Comment by KonajuGames** [comment]
The builds failed with

    CSC : error CS2001: Source file 'IGraphicsDeviceManager.SDL.cs' could not be found

The definition file lists `IGraphicsDeviceManager.SDL.cs` but the new file is `GraphicsDeviceManager.SDL.cs`.

**Comment by KonajuGames** [comment]
It will take a little while for builds and tests to run.  I'll look back on this PR in the morning (almost 1am here).

**Comment by cra0zy** [comment]
That is fine by me.

**Comment by KonajuGames** [comment]
I'm still up.  Merging now.  I'll add the unit tests from #5445 in a separate PR.

**Pull request merged**